gdk/commands/component/InitCommand.py

```diff
@@ -0,0 +1,135 @@
+            try:
+                logging.debug("Creating new project directory '{}' in the current directory.".format(project_dir.name))
+                Path(project_dir).mkdir(exist_ok=False)
+            except FileExistsError:
+                raise Exception(error_messages.INIT_DIR_EXISTS_ERROR.format(project_dir.name))
+
+        # Choose appropriate action based on commands
+        if self.command_args["template"] and self.command_args["language"]:
+            template = self.command_args["template"]
+            language = self.command_args["language"]
+            if template and language:
+                logging.info(
+                    "Initializing the project directory with a {} component template - '{}'.".format(language, template)
+                )
+                self.init_with_template(template, language, project_dir)
+                return
+        elif self.command_args["repository"]:
+            repository = self.command_args["repository"]
+            if repository:
+                logging.info(
+                    "Initializing the project directory with a component from repository catalog - '{}'.".format(repository)
+                )
+                self.init_with_repository(repository, project_dir)
+                return
+        raise Exception(error_messages.INIT_WITH_INVALID_ARGS)
+
+    def init_with_template(self, template, language, project_dir):
+        try:
+            template_name = "{}-{}".format(template, language)
+            logging.info("Fetching the component template '{}' from Greengrass Software Catalog.".format(template_name))
+            self.download_and_clean(template_name, "template", project_dir)
+        except Exception as e:
+            raise Exception("Could not initialize the project with component template '{}'.\n{}".format(template, e))
+
+    def init_with_repository(self, repository, project_dir):
+        try:
+            logging.info("Fetching the component repository '{}' from Greengrass Software Catalog.".format(repository))
+            self.download_and_clean(repository, "repository", project_dir)
+        except Exception as e:
+            raise Exception("Could not initialize the project with component repository '{}'.\n{}".format(repository, e))
+
+    def download_and_clean(self, comp_name, comp_type, project_dir):
+        """
+        Downloads component zip file file from GitHub and unarchives it in current directory.
+
+        Extracts the contents of the zip file into a temporary diretory and moves over them to the current directory.
+
+        Parameters
+        ----------
+          comp_name(string): Name of the component.
+          comp_type(string): Type of the component(template or repository).
+
+        Returns
+        -------
+            None
+        """
+        comp_url = self.get_download_url(comp_name, comp_type)
+
+        download_response = requests.get(comp_url, stream=True)
+        if download_response.status_code != 200:
```

Comment:
Is `200` enough or you need to check any other `2XX` status code? Also, how are redirects handled?

Reply:
Only `200`. We need full content to download the zip at the [URL in software catalog](https://github.com/aws-greengrass/aws-greengrass-software-catalog/blob/main/cli-components/templates.json) on the github. 

By default, `requests` module in python allows redirects for `GET`.